### PR TITLE
BigAnimal - AWS Secrets Manager Integration

### PR DESCRIPTION
**Pull request by nidhibhammar**
## What Changed?

Addresses [DOC-196](https://enterprisedb.atlassian.net/browse/DOCS-196)

### CODE REVIEW

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 1]
```diff
@@ -0,0 +1,178 @@
+---
+title: "AWS Secrets Manager integration"
+---
+
+With BigAnimal, you can use AWS Secrets Manager. AWS Secrets Manager helps you manage, retrieve, and rotate database credentials, Access keys, and other Secrets throughout their lifecycle.
```
Comment (ebgitelman) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:5]:
```suggestion
With BigAnimal, you can use AWS Secrets Manager. AWS Secrets Manager helps you manage, retrieve, and rotate database credentials, access keys, and other secrets throughout their lifecycle.
```

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+To create a secret manager:
+
+1. Create a PostgreSQL cluster on BigAnimal portal.
```
Comment (ebgitelman) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:9]:
```suggestion
1. Create a PostgreSQL cluster on the BigAnimal portal.
```

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+1. Create and save an [Access key](../reference/access_key/#create-your-personal-access-key).
```
Comment (ebgitelman) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:11]:
```suggestion
1. Create and save an [access key](../reference/access_key/#create-your-personal-access-key).
```

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+1. Create secret in application Secrets Manager for your **psql** credentials.
```
Comment (ebgitelman) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:13]:
```suggestion
1. Create a secret in AWS Secrets Manager for your psql credentials. 
```
I think this is what was meant?

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+    response = client.create_secret(
+        Name=secret_name,
+        SecretString=secret_string
+    )
+
+    return response
+    ```
+    
+    Using the created secret:
```
Comment (ebgitelman) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:43]: Does this mean: To use the created secret?

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+
+    ```shell
+    create_secret('mySecretName', 'myUsername', 'myPassword', 'myDatabase', 'myHost')
+    ```
+
+   - AWS console:
+   
+     1. Search for Secret Manager under services.
```
Comment (ebgitelman) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:51]: Should services be upper case and bold? Is it a UI element?

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+     1. Select **Store a new secret**.
+     1. Select **Credentials for other databases** on **Choose secret type** page and provide:
```
Comment (ebgitelman) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:53]:
```suggestion
     1. On the **Choose secret type** page, select **Credentials for other databases**  and provide:
```

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+        - Username
+        - Password
+        - Encryption key
+        - Database
+          Provide the server address, database name and port as per the selected database engine. Select **Next**.
```
Comment (ebgitelman) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:58]: Should this be a step? Or is this part of the Database bullet? If the latter, then Select Next should be on its own line or its own step. Couldn't make out what was intended here.

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+     1. Provide **Secret name** on the **Configure secret** page. Optionally you can provide:
```
Comment (ebgitelman) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:59]:
```suggestion
     1. On the **Configure secret** page, provide **Secret name**. Optionally, you can provide:
```

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+        - Description
+        - Tags
+        - Resource permissions
+        - Replicate secret
+        Select **Next**.
+     1. Optionally, provide details on the **Configure rotation** page.
```
Comment (ebgitelman) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:65]:
```suggestion
     1. Optionally, on the **Configure rotation** page, provide details.
```

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+     1. Review the code in different languages like: Java, JavaScript, C#, Python3, Ruby, Go and Rust. Select **Store** to create the secret manager.
```
Comment (ebgitelman) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:66]:
```suggestion
     1. Review the code in different languages like: Java, JavaScript, C#, Python3, Ruby, Go, and Rust. To create the secret manager, select **Store**.
```

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+
+1. Create secret in centralized Secrets Manager for your Access key.
```
Comment (ebgitelman) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:68]:
```suggestion
1. Create the secret in the centralized Secrets Manager for your access key.
```
Are these articles correct? Should it be "the" or "a" (in both cases)?

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+
+1. Create a sample login application:
```
Comment (ebgitelman) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:70]:
```suggestion
1. Create a sample login application.
```

Comment (ebgitelman) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:70]:
```suggestion
1. Create a sample login application.
```

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 14]
```diff
@@ -0,0 +1,178 @@
...
+
+   For example using Lambda script,
```
Comment (ebgitelman) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:72]:
```suggestion
   For example, using a Lambda script:
```

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 15]
```diff
@@ -0,0 +1,178 @@
...
+   
+   create = "create table Demo0503(Toyota int)"
+   #sql = "INSERT into secretmgr(id,name) values (%s, %s)"
+   #value = (2, "Toyota_Demo")
+   mycursor.execute(create)
+   
+   connection.commit()
+   ```
+
+  Fetch all the rows from database:
```
Comment (ebgitelman) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:107]:
```suggestion
  Fetch all the rows from the database:
```

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 16]
```diff
@@ -0,0 +1,178 @@
...
+
+   ```shell
+   print(mycursor.rowcount, "record")
+   ```
+
+1. Create Secrets Manager rotation lambda function.
```
Comment (ebgitelman) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:113]:
```suggestion
1. Create a Secrets Manager rotation Lambda function.
```
I think Lambda should be capitalized as it is elsewhere? This is the name of a programming language?

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 17]
```diff
@@ -0,0 +1,178 @@
...
+
+1. Manually execute Secrets Manager rotation script.
```
Comment (ebgitelman) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:115]:
```suggestion
1. Manually execute the Secrets Manager rotation script.
```
Should this be "a" or "the"?

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 18]
```diff
@@ -0,0 +1,178 @@
...
+
+1. Manually execute your sample application.
+
+## Example
+
+In this example a script file has all the commands required to create Secrets Manager rotation lambda function, execute the rotation script, and execute the sample application.
```
Comment (ebgitelman) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:121]:
```suggestion
In this example, a script file has all the commands required to create a Secrets Manager rotation Lambda function, execute the rotation script, and execute the sample application.
```

**Review by aaronsonntag — COMMENTED** [review]
posted a couple comments

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx (hunk 6 above)
Comment (amrita-suresh) [reply product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:51]: yes it is a UI element and should be capitalized

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx (hunk 8 above)
Comment (amrita-suresh) [reply product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:58]: it's part of the database bullet, so Select Next should be its own step

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 19]
```diff
@@ -0,0 +1,176 @@
...
+     import boto3
+     import json
+
+     def create_secret(secret_name, username, password, database, host):
+                client = boto3.client('secretsmanager')
+
+                secret_string = json.dumps({
+           "username": username,
+           "password": password,
+           "engine": "mysql",
```
Comment (nidhibhammar) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:29]:
```suggestion
           "engine": "postgresql",
```

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 20]
```diff
@@ -0,0 +1,176 @@
...
+           "host": host,
+           "dbname": database,
+           "port": 3306
```
Comment (nidhibhammar) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:32]:
```suggestion
           "port": 5432
```

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 21]
```diff
@@ -0,0 +1,176 @@
...
+   connection.commit()
+   ```
+
+  Fetch all the rows from the database:
+
+   ```shell
+   print(mycursor.rowcount, "record")
+   ```
+
+1. Create a Secrets Manager rotation Lambda function.
```
Comment (nidhibhammar) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:113]:
```suggestion
1. Create a Secrets Manager rotation Lambda function.
```

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 22]
```diff
@@ -0,0 +1,176 @@
...
+
+1. Manually execute the Secrets Manager rotation script.
```
Comment (nidhibhammar) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:115]:
```suggestion
```

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 23]
```diff
@@ -0,0 +1,176 @@
...
+
+1. Manually execute your sample application.
```
Comment (nidhibhammar) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:117]:
```suggestion
```

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 24]
```diff
@@ -0,0 +1,176 @@
...
+
+```shell
+cat gen_pass_rotate_bigani_and_secretmgr_pass.py 
+import secrets
+import string
+import requests
+import json
+import boto3
+
+key = "baak_1tbgf3PXdx2L7nclsMHAdGqHy9bNNAZtqgj3Y2DeYltjXkP5n"
```
Comment (nidhibhammar) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:131]: Do not write this exact key.

File: product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx [hunk 25]
```diff
@@ -0,0 +1,173 @@
...
```
Comment (nidhibhammar) [inline product_docs/docs/biganimal/release/administering_cluster/aws_secrets_manager_integration.mdx:128]:
```suggestion
key = "<your_access_key>"
```

**Comment by nidhibhammar** [comment]
@kwaarboncana, I made the edits as we discussed and updated the example according to your shared example. Please let me know if you need any more changes.

@amrita-suresh, Please review and let me know when to publish.

**Review by amrita-suresh — APPROVED** [review]
looks good to me! I think we are ready to publish

**Pull request merged**